app/views/hyrax/datasets/_dataset_versions.html.erb

```diff
@@ -0,0 +1,41 @@
+<h2>
+  <%= t('.header', default: 'Versions') %>
+</h2>
+
+<table class="table table-striped dataset-versions">
+  <thead>
+    <tr>
+      <th scope="col">
+	<%= t('.table_headings.version', default: 'Version') %>
+      </th>
+      <th scope="col">
+	<%= t('.table_headings.doi', default: 'DOI') %>
+      </th>
+      <th scope="col">
+	<%= t('.table_headings.provenance', default: 'Comment') %>
+      </th>
+      <th scope="col">
+	<%= t('.table_headings.publication_date', default: 'Publication Date') %>
+      </th>
+    </tr>
+  </thead>
+  <tbody>
+    <% num_versions = presenter.dataset_versions.length %>
+    <% presenter.dataset_versions.reverse.each_with_index do |ver, i| %>
+    <tr>
+      <td>
+	<%= num_versions - i %>
+      </td>
+      <td>
+	<%= link_to_unless ver.id == presenter.solr_document.id, ver.doi, ver.doi %>
+      </td>
+      <td>
+	<%= Array(ver.provenance).last %>
```

Comment:
I'm not sure Hyrax / ActiveFedora has a particularly strong (or any) notion of attribute value order.  Cf. this (admittedly old) thread -- https://groups.google.com/forum/#!searchin/samvera-tech/ordered$20attributes%7Csort:date/samvera-tech/ghqidtm_Rwo/kIhHxd0BFAAJ .  And it seems like I've heard of this topic resurfacing not all that long ago.  Fine to leave this as is ... just be aware it might not always get what you expect.  (Or maybe I'm altogether wrong about this.)  (Same applies to .first on ver.available below.)